test/common.hpp

```diff
@@ -65,6 +65,7 @@
                 __FILE__,                                                 \
                 __LINE__);                                                \
         smiErrorFlag = true;                                              \
+        return;                                                           \
```

Comment:
I don't think adding a return statement to a macro is a good idea.
This issue requires a bit more root-cause analysis to figure out why there is a hang and deal with those conditions.